vimiv/utils/imageheader.py

```diff
@@ -171,26 +184,44 @@ def _test_gif(h: bytes, _f: BinaryIO) -> bool:
     Extension: .gif
 
     Magic bytes:
-    - 47 49 46 38 37 61
-    - 47 49 46 38 39 61
+    --> 47 49 46 38 37 61
+     ->  G  I  F  8  7  a
+    --> 47 49 46 38 39 61
+     ->  G  I  F  8  9  a
 
     Support: native
     """
     return h[:4] == b"\x47\x49\x46\x38" and h[4] in [0x37, 0x39] and h[5] == 0x61
 
 
-def _test_svg(h: bytes, _f: BinaryIO) -> bool:
+def _test_svg(h: bytes, f: BinaryIO) -> bool:
     """Scalable Vector Graphics (SVG).
 
     Extension: .svg (TODO: also svgz?)
 
     Magic bytes:
-    - 3C 3F 78 6D 6C
-    - 3C 3F 73 76 67
+    --> 3C 3F 78 6D 6C (if xml, but also need to verify if svg using regex)
+     ->  <  ?  x  m  l
+    --> 3C 3F 73 76 67 (with stripped xml header)
+     ->  <  ?  s  v  g
 
     Native QT support.
     """
-    return h[:2] == b"\x3C\x3F" and (h[2:5] in [b"\x78\x6D\x6C", b"\x73\x76\x67"])
+    # Check if start with <?
+    if h[:2] != b"\x3C\x3F":
+        return False
+
+    # If stripped svg
+    if h[2:5] == b"\x73\x76\x67":
+        return True
+
+    # Check for svg start tag
+    import re
+
+    f.seek(0, 0)
+    h = f.read()
+    pattern = b"<svg"
+    return re.search(pattern, h) is not None
```

Comment:
I suppose we need the full-blown re search, as there can be all sorts of stuff in front (comments, newlines, spacing, <?xml, ...)?

Reply:
A full search opposed to a strict matching (i.e. something like `<?xml>\n<svg>`) you mean? I presume that this does not not work in all cases. I have also seen SVGs that have multiple `<svg` tag. Therefor, I thought that there may be also some other xml tags in front of it, or that the `<svg` tag is at a higher depth. But I am by no means a specialist...

Reply:
Once this is resolved, the PR is mergeable.

Reply:
Finding them is definitely a mess, I guess using `re` is the way to go, a random file I had around e.g. started with:
```
<?xml version="1.0" encoding="utf-8" standalone="no"?>
<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN"
  "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">
<!-- Created with matplotlib (https://matplotlib.org/) -->
<svg height="255pt" version="1.1" viewBox="0 0 660 255" width="660pt" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
```